api/src/test/java/org/openmrs/obs/AbstractHandlerTest.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.obs;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.openmrs.Obs;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.obs.handler.AbstractHandler;
+import org.openmrs.util.OpenmrsUtil;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ AbstractHandler.class, OpenmrsUtil.class, Context.class })
+
+public class AbstractHandlerTest {
+	
+	private final String FILENAME = "mytxtfile.txt";
+	
+	@Mock
+	private AdministrationService administrationService;
+	
+	@Rule
+	public TemporaryFolder complexObsTestFolder = new TemporaryFolder();
+	
+	@Test
+	public void shouldNeverOverwritePreviousFiles() throws IOException {
```

Comment:
@mks-d I named it the same way the other methods in the complex obs handler classes were named (the ones I didn't create) because I didn't want it to look inconsistent. Although, now that you say it, maybe these didn't have the test method name attached to it because they were not testing a particular method?

Reply:
we do still write tests the way you described it `methodName_shouldDoThisOrThat` but if the test class only tests one method I don't think you have to (unless we enforce it somewhere, I believe the `should` on the method name is enforced somewhere in our build but I am unsure, you would see that once you run `mvn clean package` I guess). (for background info on our test style evolution see  https://talk.openmrs.org/t/rethink-unit-test-style-to-remove-duplication/10200 😄 )